internal/impl/kafka/enterprise/integration_test.go

```diff
@@ -1542,3 +1542,115 @@ func TestRedpandaMigratorConsumerGroupConsistencyIntegration(t *testing.T) {
 		return false
 	}, 30*time.Second, 1*time.Nanosecond)
 }
+
+func TestRedpandaRecordOrder(t *testing.T) {
+	integration.CheckSkip(t)
+	t.Parallel()
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+	pool.MaxWait = time.Minute
+
+	source, err := startRedpanda(t, pool, true, true)
+	require.NoError(t, err)
+
+	destination, err := startRedpanda(t, pool, true, true)
+	require.NoError(t, err)
+
+	t.Logf("Source broker: %s", source.brokerAddr)
+	t.Logf("Destination broker: %s", destination.brokerAddr)
+
+	dummyTopic := "foobar"
+	dummyMessage := `{"test":"foo"}`
+	go func() {
+		t.Log("Producing messages...")
+
+		produceMessages(t, source, dummyTopic, dummyMessage, 0, 50, false, 50*time.Millisecond)
+
+		t.Log("Finished producing messages")
+	}()
+
+	runRedpandaPipeline := func(t *testing.T, source, destination redpandaEndpoints, topic string, suppressLogs bool) {
+		streamBuilder := service.NewStreamBuilder()
+		require.NoError(t, streamBuilder.SetYAML(fmt.Sprintf(`
+input:
+  redpanda:
+    seed_brokers: [ %s ]
+    topics: [ %s ]
+    consumer_group: migrator_cg
+    start_from_oldest: true
+
+output:
+  redpanda:
+    seed_brokers: [ %s ]
+    topic: ${! @kafka_topic }
+    key: ${! @kafka_key }
```

Comment:
The original produced data does mot have a key set, so it goes to a random partition. We will ensure order only because there is a single partition then right? I know we auto create topics in this test, I am assuming those are created with a single partition otherwise I am confused how this works

Reply:
I just pushed a change to force a single partition. Also, I do set the key to `${! counter() }` [here](https://github.com/redpanda-data/connect/blob/35ed595740a20b29bd3e66f2ef73d415fbc6de62/internal/impl/kafka/enterprise/integration_test.go#L847) so we should have deterministic keys.